### PR TITLE
Add back Notebook Completion List IntelliSense

### PR DESCRIPTION
**Pull request by chlafreniere**
Add back Notebook Completion List IntelliSense.

Implements #3223.

### CODE REVIEW

File: src/sql/parts/notebook/notebook.component.ts [hunk 1]
```diff
@@ -207,7 +207,11 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 	private async doLoad(): Promise<void> {
 		try {
 			await this.loadModel();
+			// this.notebookService.addNotebookEditor(this);
```
Comment (YurongHe) [inline src/sql/parts/notebook/notebook.component.ts]: Dead code to be removed?

File: src/sql/parts/notebook/notebook.component.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 	ngOnInit() {
 		this._register(this.themeService.onDidColorThemeChange(this.updateTheme, this));
 		this.updateTheme(this.themeService.getColorTheme());
-		this.notebookService.addNotebookEditor(this);
+		// this.notebookService.addNotebookEditor(this);
```
Comment (YurongHe) [inline src/sql/parts/notebook/notebook.component.ts]: No need?

File: src/sql/workbench/api/node/extHostNotebookDocumentsAndEditors.ts [hunk 3]
```diff
@@ -136,6 +138,21 @@ export class ExtHostNotebookDocumentsAndEditors implements ExtHostNotebookDocume
...
 	}
+
+	$acceptModelChanged(uriComponents: UriComponents, e: INotebookModelChangedData): void {
+		const uri = URI.revive(uriComponents);
+		const strURL = uri.toString();
+		let data = this._documents.get(strURL);
+		if (data) {
+			data.onModelChanged(e);
+		}
+		this._onDidChangeNotebookCell.fire({
```
[the remark this answers is not in the document]
Comment (chlafreniere) [reply src/sql/workbench/api/node/extHostNotebookDocumentsAndEditors.ts]: Yup, good catch. Thanks.

File: src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts [hunk 4]
```diff
@@ -407,8 +445,54 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
...
+		else {
+			notebookCells.push({
+				uri: cells.cellUri,
+				contents: {
+					cell_type: cells.cellType,
+					execution_count: undefined,
+					metadata: {
+						language: cells.language
+					},
+					source: undefined
```
[the remark this answers is not in the document]
Comment (chlafreniere) [reply src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts]: #3527

**Pull request merged**